lib/stage0-handoff/src/lib.rs

```diff
@@ -0,0 +1,128 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#![cfg_attr(not(test), no_std)]
+
+use core::convert::From;
+use core::marker::Sized;
+use core::ops::Range;
+use hubpack::SerializedSize;
+use serde::{Deserialize, Serialize};
+use static_assertions::const_assert;
+
+mod rot_update_details;
+
+pub use rot_update_details::{
+    ImageVersion, RotBootState, RotImageDetails, RotSlot,
+};
+
+// This memory is the USB peripheral SRAM that's 0x4000 bytes long. Changes
+// to this address must be coordinated with the [dice_*] tables in
+// chips/lpc55/chip.toml
+// TODO: get from app.toml -> chip.toml at build time
```

Comment:
Do you want to do this as part of this PR?

Reply:
TLDR: Let's leave it as is for this PR.

Longer answer:

This is a left over TODO from @flihp. However, as we've discovered alignment of 16 bits is required when the USB SRAM is treated as a peripheral, and that doesn't always play nicely with hubpack serialization and compiler optimizations. This PR works around that by setting the USB SRAM region as normal memory in the MPU in stage0, which has the effect of removing alignment requirements. I then took the advice of @cbiffle and @labbott and made the mapping in hubris its own linker section and pulled it in as a static with `#[link_section = ".bootstate]`. That actually already does fix the issue for my bootstate code here. The `UPDATE_RANGE` no longer lives in `chip.toml`, but in `memory.toml` and `256k.toml` and no hardcoding of addresses is used. This particular constant is left over to show what ranges exist while we still have things in `chip.toml`. 

All that being said, I *think* we'll want to do the same thing for the `DICE_RANGE` sections in chip.toml, but I'm not 100% sure. That's all @flihp's code and I don't want to make that change without him.

Reply:
No objections here. I'll get on changing the way the DICE regions are mapped once this is merged.